raiden-ts/src/config.ts

```diff
@@ -0,0 +1,32 @@
+import * as t from 'io-ts';
+
+/**
+ * An Raiden configuration object to overwrite the Raiden default configuration.
+ * 
+ * @matrixServer Matrix server to initliaze the [[Raiden]] instance with.
+ * The matrix server can't be changed later on.
+ */
+export const PartialRaidenConfig = t.partial({
+  matrixServer: t.string,
+});
+
+/**
+ * A Raiden configuration object with required parameters and
+ * optional parameters from [[PartialRaidenConfig]].
+ * 
+ * @matrixServerLookup Matrix server URL to fetch existing matrix servers from. 
+ * After intializing a [[Raiden]] instance, the matrix server can't be changed later on.
+ * @revealTimeout Timeout for secrets to be revealed
+ * @settleTimeout Timeout for channels to be settled
+ */
+export const RaidenConfig = t.intersection([
+  t.type({
+    matrixServerLookup: t.string,
+    revealTimeout: t.number,
+    settleTimeout: t.number,
+  }),
+  PartialRaidenConfig,
+]);
+
+export type RaidenConfigDefaults = t.TypeOf<typeof RaidenConfig>;
+export type RaidenConfig = Partial<RaidenConfigDefaults>;
```

Comment:
Don't need to export the Partial version, just the normal codec and type. In the few places it'll be needed (raiden factory, reset function), it's actually more descriptive to explicitly have `Partial<RaidenConfig>` there, as it informs the developer it accepts an object containing a subset of the configuration type.